### ISSUE TITLE
Symbols, use cases

### ISSUE THREAD

**Issue by benlesh** [DrkSephy/es6-cheatsheet#32]
This repo went around Netflix. After reading it there were a few people perhaps confused about Symbols. This is a copy-paste of my response there...

---

Symbol is useful for when you want to put a key on an object without trampling other keys on that object. Each created Symbol is immutable and completely unique.

Let's say I have code that wants to patch `React` with some namespace `Foo`. (I have no idea why I want to do this, but i do)... and I don't want the code to break if the good folks at Facebook want to add a `Foo` namespace themselves:

```
// this will break when they update.
React.Foo = { my: 'stuff' };

// there is no way they can break this
// because they can't possibly create the Foo symbol themselves
const Foo = Symbol(); // notice Symbol and not Symbol.for
React[Foo] = { my: 'stuff' };
```

Another use is for interoperability points...

Say I have some library, `Netflixer`. I can say "anything that has a method at key `FlixIt`, which is a Symbol I'm sharing, I know I can read". In order to "share" a Symbol, you put it in a global registry by using `Symbol.for(key)` to create or retrieve it.

So the third party that wants interop can do this:

```
myType[Symbol.for('FlixIt')] = (x) => 'myType flixed by ' + x;
```

and in our Netflixable type:

```
Netflixer.flix = (obj) => {
  const flixIt = obj[Symbol.for('FlixIt')];
  if (flixIt) {
    flixIt('weee');
  }
}
```

A notable example of `Symbol` usage is `Symbol.iterator` which should exist on any Iterable or Iterator type (Array, string, generators, etc)

**Comment by DrkSephy** [comment]
@blesh, thank you so much for your comments on this. As mentioned in the Symbols example in the readme, they are useful for making sure that keys don't collide inside of an object. However, I agree that the example I created doesn't do a good a job of highlighting this. 

If you don't mind, would you be willing to submit a PR with the example you've provided here? It highlights the point clearly.  As for your second example, that is much appreciated too - I had considered having an example of storing symbols and using them in such a matter, but once again you've done a great job and I'd love to have these examples in the readme.

**Comment by benlesh** [comment]
Sure. I'll try to put something together tomorrow.

**Comment by DrkSephy** [comment]
Awesome, much appreciated!

**Comment by DrkSephy** [comment]
Closed by commit d8a45f2073f128e968dcf1b5507027fb34cf4d88.
